File: csharp/07 - Demo 3 - Refactoring/Refactoring/IntroduceParameterObject/Account.cs
using System;
using System.Collections.Generic;

namespace Refactoring.IntroduceParameterObject
{
    public class Account
    {
        public ISet<Entry> Entries { get; } = new HashSet<Entry>();

        public double GetFlowBetween(DateTime start, DateTime end)
        {
            double result = 0;
            foreach (var entry in Entries)
            {
                if (entry.ChargeDate == start
                        || entry.ChargeDate == end
                        || (entry.ChargeDate > start && entry.ChargeDate < end))
                {
                    result += entry.Value;
                }
            }

            return result;
        }

    }
}

File: java/07 - Demo 3 - Refactoring/src/main/java/introduceparameterobject/Account.java
package introduceparameterobject;

import java.util.Date;
import java.util.HashSet;
import java.util.Set;

public class Account {

	private Set<Entry> entries = new HashSet<Entry>();

	Account() {
	}

	double getFlowBetween(Date start, Date end) {
		double result = 0;
		for (Entry each : entries) {
			if (each.getDate().equals(start)
					|| each.getDate().equals(end)
					|| (each.getDate().after(start) && each.getDate().before(end))) {
				result += each.getValue();
			}
		}
		return result;
	}

	Set<Entry> getEntries() {
		return entries;
	}

	void setEntries(Set<Entry> entries) {
		this.entries = entries;
	}
	
}
